doc/over_react_redux_documentation.md

```diff
@@ -275,6 +279,270 @@ UiFactory<CounterProps> Counter = connect<CounterState, CounterProps>(
+use [`createStoreHook`](#custom-context-for-hooks) instead.
+
+> See the [react-redux JS documentation](https://react-redux.js.org/api/hooks#usestore) for more details.
+
+### Custom Context For Hooks
+The [`ReduxProvider`](#reduxprovider) component allows you to specify an alternate `Context` via `props.context`. 
+This is useful if you're building a complex reusable component, and you don't want your store to collide with any 
+Redux store your consumers' applications might use, or you're using [multiple Redux stores](#using-multiple-stores) 
+in your application.
+
+To access an alternate context via the hooks API, use the hook creator functions:
+
+```dart
+import 'package:over_react/over_react.dart';
+import 'package:over_react/over_react_redux.dart';
+import 'package:redux/redux.dart';
+
+// ------------------------------------
+//  1. Declare the custom context
+// ------------------------------------
+final MyContext = createContext();
+final useSelector = createSelectorHook<MyState>(MyContext);
+final useDispatch = createDispatchHook(MyContext);
+// This should probably not be used frequently. Prefer `createSelectorHook` as your primary choice.
+// However, this may be useful for less common scenarios that do require access to the `Store`,
+// such as replacing reducers.
+final useStore = createStoreHook<MyState>(MyContext);
+
+final myStore = Store(myReducer);
+
+// ------------------------------------
+//  2. Create a function component that
+//  uses the shadowed `useSelector`.
+// ------------------------------------
+mixin MyComponentProps on UiProps {}
+
+UiFactory<MyComponentProps> MyComponent = uiFunction(
+  (props) {
+    final count = useSelector((state) => state.count);
+    // This should probably not be used frequently. Prefer `createSelectorHook` as your primary choice.
+    // However, this may be useful for less common scenarios that do require access to the `Store`,
+    // such as replacing reducers.
+    final store = useStore();
+    final dispatch = useDispatch();
+
+    return Dom.div()('The current count is $count');
+  },
+  $MyComponentConfig, // ignore: undefined_identifier
+);
+
+// ------------------------------------
+//  3. Render the function component
+//  nested within the ReduxProvider
+//  that is wired up to the 
+//  custom context / store.
+// ------------------------------------
+main() {
+  final app = (ReduxProvider()
+    ..context = MyContext
+    ..store = myStore
+  )(
+    MyComponent()(),
+  );
+
+  react_dom.render(app, querySelector('#id_of_mount_node'));
+}
+```
+
+> __CAUTION:__ Be sure to not export the shadowed values of `useSelector`, `useDispatch` or `useStore` unless you know 
+  exactly what you're doing, and the consumers of your library expect the hook to always have the context of the 
+  state you parameterize it with.
+
+> See the [react-redux JS documentation](https://react-redux.js.org/api/hooks#custom-context) for more details.
+
+
 ## Using Multiple Stores
```

Comment:
Can we update this section to also mention / link to the "Custom Context For Hooks" section for consumers using hooks?